src/Common/src/System/Runtime/InteropServices/StringBuffer.cs

```diff
@@ -76,22 +76,24 @@ public ulong CharCapacity
         /// Ensure capacity in characters is at least the given minimum.
         /// </summary>
         /// <exception cref="OutOfMemoryException">Thrown if unable to allocate memory when setting.</exception>
-        public void EnsureCharCapacity(ulong minCapacity)
+        public void EnsureCharCapacity(uint minCapacity)
         {
-            EnsureByteCapacity(minCapacity * sizeof(char));
+            EnsureByteCapacity(minCapacity * (ulong)sizeof(char));
         }
 
         /// <summary>
         /// The logical length of the buffer in characters. (Does not include the final null.) Will automatically attempt to increase capacity.
         /// This is where the usable data ends.
         /// </summary>
-        /// <exception cref="ArgumentOutOfRangeException">Thrown if attempting to set <paramref name="nameof(Length)"/> to a value that is larger than the maximum addressable memory.</exception>
         /// <exception cref="OutOfMemoryException">Thrown if unable to allocate memory when setting.</exception>
-        public unsafe ulong Length
+        /// <exception cref="ArgumentOutOfRangeException">Thrown if the set size in bytes is uint.MaxValue (as space is implicitly reservced for the trailing null).</exception>
+        public unsafe uint Length
         {
             get { return _length; }
             set
             {
+                if (value == uint.MaxValue) throw new ArgumentOutOfRangeException("Length");
```

Comment:
Good to know. For some reason I thought you couldn't do properties. I'll fix it.